python/paddle/fluid/tests/unittests/test_linalg_lstsq_op.py

```diff
@@ -0,0 +1,212 @@
+            self.dtype)
+
+    def generate_output(self):
+        if len(self._input_shape_1) == 2:
+            out = np.linalg.lstsq(
+                self._input_data_1, self._input_data_2, rcond=self.rcond)
+        elif len(self._input_shape_1) == 3:
+            out = np.linalg.lstsq(
+                self._input_data_1[0], self._input_data_2[0], rcond=self.rcond)
+
+        self._output_solution = out[0]
+        self._output_residuals = out[1]
+        self._output_rank = out[2]
+        self._output_sg_values = out[3]
+
+    def test_dygraph(self):
+        paddle.disable_static()
+        paddle.device.set_device("cpu")
+        place = paddle.CPUPlace()
+        x = paddle.to_tensor(self._input_data_1, place=place, dtype=self.dtype)
+        y = paddle.to_tensor(self._input_data_2, place=place, dtype=self.dtype)
+        results = paddle.linalg.lstsq(
+            x, y, rcond=self.rcond, driver=self.driver)
+
+        res_solution = results[0].numpy()
+        res_residuals = results[1].numpy()
+        res_rank = results[2].numpy()
+        res_singular_values = results[3].numpy()
+
+        if x.shape[-2] > x.shape[-1] and self._output_rank == x.shape[-1]:
+            if (np.abs(res_residuals - self._output_residuals) < 1e-6).any():
+                pass
+            else:
+                raise RuntimeError("Check LSTSQ residuals dygraph Failed")
+
+        if self.driver in ("gelsy", "gelsd", "gelss"):
+            if (np.abs(res_rank - self._output_rank) < 1e-6).any():
+                pass
+            else:
+                raise RuntimeError("Check LSTSQ rank dygraph Failed")
+
+        if self.driver in ("gelsd", "gelss"):
+            if (np.abs(res_singular_values - self._output_sg_values) < 1e-6
+                ).any():
+                pass
+            else:
+                raise RuntimeError("Check LSTSQ singular values dygraph Failed")
+
+    def test_static(self):
+        paddle.enable_static()
+        paddle.device.set_device("cpu")
+        place = fluid.CPUPlace()
+        with fluid.program_guard(fluid.Program(), fluid.Program()):
+            x = paddle.fluid.data(
+                name="x",
+                shape=self._input_shape_1,
+                dtype=self._input_data_1.dtype)
+            y = paddle.fluid.data(
+                name="y",
+                shape=self._input_shape_2,
+                dtype=self._input_data_2.dtype)
+            results = paddle.linalg.lstsq(
+                x, y, rcond=self.rcond, driver=self.driver)
+            exe = fluid.Executor(place)
+            fetches = exe.run(
+                fluid.default_main_program(),
+                feed={"x": self._input_data_1,
+                      "y": self._input_data_2},
+                fetch_list=[results])
+
+            if x.shape[-2] > x.shape[-1] and self._output_rank == x.shape[-1]:
+                if (np.abs(fetches[1] - self._output_residuals) < 1e-6).any():
```

Comment:
use np.allclose instead of np.abs and any ? generally, set rtol/atol to 1e-6/0
the same as other np.abs below

Reply:
Thanks, I will fix them in next PR.